python/lsst/pipe/analysis/plotRhoStatistics.py

```diff
@@ -0,0 +1,347 @@
+        
+def plotEllipResCorrVisits(butler, visits, filter, filename):
+    fig, axes  = plt.subplots(1,2)
+    fig.set_size_inches(16,9)
+
+    shapeCol = "ext_shapeHSM_HsmSourceMoments"
+    psfShapeCol = "ext_shapeHSM_HsmPsfMoments"
+    e1Func = E1(shapeCol)
+    e2Func = E2(shapeCol)
+    e1ResidsFunc = E1Resids(shapeCol, psfShapeCol)
+    e2ResidsFunc = E2Resids(shapeCol, psfShapeCol)
+    TraceSizeResidsFunc = PsfTraceSizeDiff(shapeCol, psfShapeCol)
+
+    # Holders for multiple visits
+    radiiAll = {k:[] for k in range(1,6) }
+    xipAll = {k:[] for k in range(1,6) }
+    xipErrAll = {k:[] for k in range(1,6) }
+
+    for visit in visits:
+        # print("visit = {}".format(visit))
+        e1, e2 = [], []
+        e1Res, e2Res = [], []
+        SizeRes = []
+        ra = []
+        dec = []
+        notFoundCcds = []
+        for i in range(104):
+            dataId = {"visit": visit, "ccd": i, "filter": filter}
+            try:
+                catalog = butler.get('src', dataId, immediate=True, flags=afwTable.SOURCE_IO_NO_FOOTPRINTS)
+            except:
+                notFoundCcds.append(i)
+                continue
+            psfUsedCat = catalog[catalog["calib_psf_used"]].copy(deep=True)
+
+            e1CCD = e1Func(psfUsedCat)
+            e2CCD = e2Func(psfUsedCat)
+            e1Residuals = e1ResidsFunc(psfUsedCat)
+            e2Residuals = e2ResidsFunc(psfUsedCat)
+            SizeResiduals = TraceSizeResidsFunc(psfUsedCat)/100. # output is in percentage
+
+            ras = np.degrees(psfUsedCat["coord_ra"])
+            decs = np.degrees(psfUsedCat["coord_dec"])
+            
+            isFinite = np.isfinite(e1Residuals) & np.isfinite(e2Residuals) & np.isfinite(SizeResiduals)
+            
+            e1CCD = e1CCD[isFinite]
+            e2CCD = e2CCD[isFinite]
+            e1Residuals = e1Residuals[isFinite]
+            e2Residuals = e2Residuals[isFinite]
+            SizeResiduals = SizeResiduals[isFinite] 
+            ras = ras[isFinite]
+            decs = decs[isFinite]
+            
+            if np.isnan(np.mean(e1Residuals)) or np.isnan(np.mean(e2Residuals)) or np.isnan(np.mean(SizeResiduals)):
+                # print("****  Error computing e1Res and/or e2Res for ccd: {}  ****".format(patch))
+                continue
+
+            e1.extend(e1CCD)
+            e2.extend(e2CCD)
+            e1Res.extend(e1Residuals)
+            e2Res.extend(e2Residuals)
+            SizeRes.extend(SizeResiduals)
+            ra.extend(ras)
+            dec.extend(decs)
+
+        # print("List of ccds NOT found: {}".format(notFoundCcds))
+        # print("mean e1Res: {}, mean e2Res: {}".format(np.mean(e1Res), np.mean(e2Res)))
+
+        e1 = np.array(e1)
+        e2 = np.array(e2)
+        SizeRes = np.array(SizeRes)
+        e1Res = np.array(e1Res)
+        e2Res = np.array(e2Res)
+        
+        treecorr_kwargs = {'nbins':10, 'min_sep':0.5, 'max_sep':10, 'sep_units':"arcmin", 'verbose':0}
```

Comment:
Case inconsistencies

Reply:
I could change the case for treecorr_kwargs, but I cannot change the cases for the keys themselves, e.g., min_sep, because that's what treecorr expects.

Reply:
The max_sep value is smaller since this is expected to be routinely run on three disjoint tracts, and not a contiguous area. I'll make the value configurable, but the maximum separation still has to be smaller than the scale of a tract.